File: yak-gl-api/src/main/java/net/yakclient/opengl/api/gui/SpecifiedProperties.java
package net.yakclient.opengl.api.gui;

import java.util.Optional;

/**
 * Defines pre-determined properties that given GUIProperties
 * should have.
 *
 * @author Durgan McBroom
 */
public abstract class SpecifiedProperties {
    private final GUIProperties properties;

    public SpecifiedProperties(GUIProperties properties) {
        this.properties = properties;
    }

    protected <T> Optional<T> get(String name) {
        return Optional.ofNullable(this.properties.get(name));
    }
}

File: api/src/main/kotlin/net/yakclient/graphics/api/gui/SpecifiedProperties.kt
package net.yakclient.graphics.api.gui

import java.util.Optional

/**
 * Defines pre-determined properties that given GUIProperties
 * should have.
 *
 * @author Durgan McBroom
 */
abstract class SpecifiedProperties(private val properties: GuiProperties) {
    protected operator fun <T> get(name: String): Optional<T> {
        return Optional.ofNullable(properties.get(name))
    }
}